tickit/adapters/interpreters/wrappers/recoding_interpreter.py

```diff
@@ -0,0 +1,85 @@
+from typing import AnyStr, AsyncIterable, Tuple, cast
+
+from tickit.core.adapter import Adapter, Interpreter
+
+
+class DecodingInterpreter(Interpreter[AnyStr]):
+    """A wrapper for an interpreter that decodes messages.
+
+    An interpreter wrapper class that recieves messages, decodes them, and then passes
+    them on to the wrapped interpreter to handle.
+    """
+
+    def __init__(
+        self, interpreter: Interpreter[AnyStr], encoding: str = "utf-8"
+    ) -> None:
+        """A wrapper for an interpreter that decodes messages.
+
+        Args:
+            interpreter (Interpreter): The interpreter decoded messages are passed to.
+            encoding (str): The encoding with which to decode the incoming message.
+        """
+        self.interpreter: Interpreter[AnyStr] = interpreter
+        self.encoding = encoding
+        super().__init__()
+
+    async def handle(
+        self, adapter: Adapter, message: AnyStr
+    ) -> Tuple[AsyncIterable[AnyStr], bool]:
```

Comment:
This should be typed as:

```suggestion
    async def handle(
        self, adapter: Adapter, message: bytes
    ) -> Tuple[AsyncIterable[str], bool]:
```

Though it may require us to use a contravarient version of `AnyStr` in the `Interpreter` definition